core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -478,7 +481,8 @@ class ReplicaManager(val config: KafkaConfig,
               if (requestLeaderEpoch == LeaderAndIsr.EpochDuringDelete ||
                   requestLeaderEpoch == LeaderAndIsr.NoEpoch ||
                   requestLeaderEpoch >= currentLeaderEpoch) {
-                stoppedPartitions += topicPartition -> deletePartition
+                stoppedPartitions += StopPartition(topicPartition, deletePartition,
+                  deletePartition && partition.isLeader && requestLeaderEpoch == LeaderAndIsr.EpochDuringDelete)
```

Comment:
This may be fine for now. However, the edge case is that it's possible for a partition to be already leaderless (for example, no replica in ISR) when a topic is deleted. Could we file a jira to track this?

Reply:
Filed KAFKA-15331 to track this case.